native/lib.rs

```diff
@@ -0,0 +1,45 @@
+mod droptimer;
+
+use helium_crypto::{ecc_compact, ed25519, KeyType, Verify};
+use rustler::{Atom, Binary, Error as NifError, ListIterator, NifResult};
+use std::convert::TryFrom;
+
+// -spec verify(
+//     [{Bin :: binary(), [{Signature :: binary(), PubKeyBin :: libp2p_crypto:pubkey_bin()}, ...]}]
+// ) -> ok | {error, Reason :: binary()}.
+#[rustler::nif(name = "verify", schedule = "DirtyCpu")]
+fn verify_1(batch: ListIterator) -> NifResult<Atom> {
+    // let _timer = droptimer::DropTimer::new();
+    for sub_batch in batch.map(|term| term.decode::<(Binary, ListIterator)>()) {
+        let (msg, list) = sub_batch?;
+        for pair in list.map(|term| term.decode::<(Binary, Binary)>()) {
+            let (signature, pubkey_bin) = pair?;
```

Comment:
Can you make it require the pubkey_bins per Binary that's being signature checked are unique? As in you expect only one instance of the pubkeybin per sub_batch

Reply:
Yep. It may complicate things if we want to add parallelism. And of course it’ll be a slight perf hit